falcon/api.py

```diff
@@ -116,13 +120,25 @@ def process_response(self, req, resp, resource)
 
     def __init__(self, media_type=DEFAULT_MEDIA_TYPE, before=None, after=None,
                  request_type=Request, response_type=Response,
-                 middleware=None, router=None):
+                 middleware=None, router=None, cors=None):
         self._sinks = []
         self._media_type = media_type
 
         self._before = helpers.prepare_global_hooks(before)
         self._after = helpers.prepare_global_hooks(after)
 
+        if cors is not None:
+            if not isinstance(cors, CORS):
+                raise ValueError('cors argument must be an instance of '
+                                 'falcon.cors.CORS')
+            cors_middleware = CORSMiddleware(cors)
+            if type(middleware) == list:
```

Comment:
its recommended to use isinstance:
if not isinstance(middleware, list):
    ....